File: 0078-subsets/0078-subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        n = len(nums)
        ans = []
        for mask in range(1 << n):
            sub = []
            for i in range(n):
                if mask >> i & 1:
                    sub.append(nums[i])
            ans.append(sub)
        return ans

File: 0078-subsets/0078-subsets.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var subsets = function(nums) {
  const n = nums.length;
  const ans = [];
  
  for (let mask = 0; mask < 1 << n; mask++) {
    const sub = [];
    for (let j = 0; j < n; j++) {
      if (mask >> j & 1) sub.push(nums[j]); 
    }
    
    ans.push(sub);
  }
  
  return ans;
};